Dashboard/pages/price.py

```diff
@@ -0,0 +1,57 @@
+"""Price tracker page for streamlit"""
+
+import logging
+
+import streamlit as st
+from streamlit.delta_generator import DeltaGenerator
+
+from navigation import make_sidebar
+
+
+def price_tracker_page() -> None:
+    """Product Price Tracker Page"""
+    st.title("Product Price Tracker")
+
+    with st.expander("List of Compatible websites", expanded=True):
+        st.write("ASOS")
+
+    product_url = st.text_input("Enter the product URL:")
+
+    if st.button("Track Price", type="primary"):
+        if product_url:
+            subscribe(product_url)
+        else:
+            st.error("Please enter a valid product URL.")
+
+
+def subscribe(product_url: str) -> DeltaGenerator:
```

Comment:
What is the `DeltaGenerator` type hint?

Reply:
no idea but it's the one that the function wanted